### PR TITLE
remove removal schedule of http_uri and change its implementation to uri_string

### PR DESCRIPTION
**Pull request by Whaileee**

### CODE REVIEW

File: lib/inets/src/http_lib/http_uri.erl [hunk 1]
```diff
@@ -39,58 +39,8 @@
 %%%=========================================================================
 %%%  API
 %%%=========================================================================
-reserved() ->
-    sets:from_list([$;, $:, $@, $&, $=, $+, $,, $/, $?,
-            $#, $[, $], $<, $>, $\", ${, $}, $|, %"
-			       $\\, $', $^, $%, $ ]).
+encode(URI) ->
```
[the remark this answers is not in the document]
Comment (u3s) [reply lib/inets/src/http_lib/http_uri.erl:42]: maybe even rely more on uri_string:quote? use more general term `Data`.
https://www.erlang.org/doc/man/uri_string#quote-1

URIPart is a bit fuzzy ... one may think path is a part of URI and be disappointed with function behavior?

**Review by u3s — COMMENTED** [review]
I think it would be good to merge this PR today or tomorrow so it is included in RC2.

as for this part of decision: _"leave docs as but make it more visible that it is deprecated and make sure to link to uri_string functions"_

I think the link to uri_string is already present and _"make it more visible"_ part can be discussed after RC2 and adjusted later (preferably aligned with approach in other apps - ssl?)

File: lib/inets/src/http_lib/http_uri.erl (hunk 1 above)
[the remark this answers is not in the document]
Comment (IngelaAndin) [reply lib/inets/src/http_lib/http_uri.erl:42]: Maybe Fragment would be better? URI is not good!

**Comment by IngelaAndin** [comment]
> I think it would be good to merge this PR today or tomorrow so it is included in RC2.
> 
> as for this part of decision: _"leave docs as but make it more visible that it is deprecated and make sure to link to uri_string functions"_
> 
> I think the link to uri_string is already present and _"make it more visible"_ part can be discussed after RC2 and adjusted later (preferably aligned with approach in other apps - ssl?)

I agree that documentation may be improved after RC2. Yes the link is in the docs, but the comment formulation is not adequate for  what was decided on OTB, and of course we need specs for OTP-27 release.

File: lib/inets/src/http_lib/http_uri.erl (hunk 1 above)
[the remark this answers is not in the document]
Comment (essen) [reply lib/inets/src/http_lib/http_uri.erl]: Fragment refers specifically to the `#fragment` part of a URI so it shouldn't be used for this. `Data` is fine assuming the function is general.

[the remark this answers is not in the document]
Comment (IngelaAndin) [reply lib/inets/src/http_lib/http_uri.erl]: Yes, this function will only be kept for backwards compatibility and will now be a wrapper for uri_string:encode/decode which are the recommended functions to use (that is http_uri functions  will no longer be bug compatible), which calls it  Data so lets go for that.

[the remark this answers is not in the document]
Comment (Whaileee) [reply lib/inets/src/http_lib/http_uri.erl]: Okay, so Data it is

File: lib/inets/src/http_lib/http_uri.erl [hunk 2]
```diff
@@ -56,80 +56,37 @@ For more information about URI, see
...
 -doc """
 encode(DecodedPart) -> EncodedPart
 
 > #### Warning {: .warning }
 >
-> Do not use will be removed use `uri_string:quote/1` instead
+> Use `uri_string:quote/1` instead
 
-Performes prrcent encoding.
+Performes percent encoding.
```
Comment (u3s) [inline lib/inets/src/http_lib/http_uri.erl:67]: typo: Performes -> Performs

**Review by u3s — APPROVED** [review]

**Review by u3s — APPROVED** [review]

**Pull request merged**